### ISSUE TITLE
The space disappears from the code block in the dictionary list item.

### ISSUE THREAD

**Issue by sugitach** [jgm/pandoc#4051]
In execute pandoc in command line.

~~~
% pandoc
test1
:    test
```json
{
  "a":"b"
  [
    {
      "x":"x"
    }
  ]
}
```
^D
<dl>
<dt>test1</dt>
<dd><p>test</p>
<pre class="sourceCode json" id="cb1"><code class="sourceCode json"><div class="sourceLine" id="cb1-1" data-line-number="1"><span class="fu">{</span></div>
<div class="sourceLine" id="cb1-2" data-line-number="2">  <span class="dt">&quot;a&quot;</span><span class="fu">:</span><span class="st">&quot;b&quot;</span></div>
<div class="sourceLine" id="cb1-3" data-line-number="3">  <span class="ot">[</span></div>
<div class="sourceLine" id="cb1-4" data-line-number="4"><span class="fu">{</span></div>
<div class="sourceLine" id="cb1-5" data-line-number="5">  <span class="dt">&quot;x&quot;</span><span class="fu">:</span><span class="st">&quot;x&quot;</span></div>
<div class="sourceLine" id="cb1-6" data-line-number="6"><span class="fu">}</span></div>
<div class="sourceLine" id="cb1-7" data-line-number="7">  <span class="ot">]</span></div>
<div class="sourceLine" id="cb1-8" data-line-number="8"><span class="fu">}</span></div></code></pre>
</dd>
</dl>
~~~

Four consolidated white spaces have disappeared.
But, normal list items has not that.

~~~
% pandoc
* test
```json
{
  "a":"b"
  [
    {
      "x":"x"
    }
  ]
}
```
^D
<ul>
<li>test</li>
</ul>
<pre class="sourceCode json" id="cb1"><code class="sourceCode json"><div class="sourceLine" id="cb1-1" data-line-number="1"><span class="fu">{</span></div>
<div class="sourceLine" id="cb1-2" data-line-number="2">  <span class="dt">&quot;a&quot;</span><span class="fu">:</span><span class="st">&quot;b&quot;</span></div>
<div class="sourceLine" id="cb1-3" data-line-number="3">  <span class="ot">[</span></div>
<div class="sourceLine" id="cb1-4" data-line-number="4">    <span class="fu">{</span></div>
<div class="sourceLine" id="cb1-5" data-line-number="5">      <span class="dt">&quot;x&quot;</span><span class="fu">:</span><span class="st">&quot;x&quot;</span></div>
<div class="sourceLine" id="cb1-6" data-line-number="6">    <span class="fu">}</span></div>
<div class="sourceLine" id="cb1-7" data-line-number="7">  <span class="ot">]</span></div>
<div class="sourceLine" id="cb1-8" data-line-number="8"><span class="fu">}</span></div></code></pre>
~~~

**Comment by jgm** [comment]
You should indent block-level content in a definition list 4 spaces.
It happens to work, sort of, without this, because we allow "laziness."  But laziness is intended for ordinary paragraph continuation lines, not new block-level content.

We should fix the definition list parser, though, so it stops the list when you hit the fenced code block, as with the bullet list.

**Comment by sugitach** [comment]
Hi, jgm.
I caught same case in indented bullet list.

~~~
* top level.
    * some line on indented level.
      ```json
{
  "test":"code block in 1 leve right indenting",
    "test":"code block in 2 leve right indenting",
      "test":"code block in 1 leve right indenting",
}
```
* top level again.
~~~

Is there any way to indent code blocks somehow?
Trying to indent code blocks as below does not work.

~~~
* some line on top level.

    ```json
    {
      "test":"code block in 1 leve left indenting"
    }
    ```

* top level again.
~~~

**Comment by jgm** [comment]
> Is there any way to indent code blocks somehow?

They must be indented four spaces beyond the column where the first non-space occurs after the list marker (in your case, the `s` on the first line).

**Comment by jgm** [comment]
The fence must start in the same column as the first non-space character after the bullet (the `s` in line 1).

**Comment by jgm** [comment]
We might want to relax this, as CommonMark allows these to be indented.

**Comment by sugitach** [comment]
Thank you. I got it.

~~~
Dict1
:   

    ```json
    {
      "a":"a",
        "b":"b",
          "c":"c"
    }
    ```
~~~

But, only at 4 spaces  each line is succeeded.

* first ":" + 3 spaces line and 4 spaces each line --- success
* first ":" + 2 spaces line and 3 spaces each line -- fail 
* first ":" + 4 spaces line and 5 spaces each line -- fail

such a thing?

**Comment by jgm** [comment]
Closed by commit 1592d3882142bbb938608e04d179f148453d93bb

**Comment by jgm** [comment]
I've pushed a commit that makes pandoc more forgiving about
indentation here.
